### PR TITLE
Set channel pool size at creation time.

### PR DESCRIPTION
**Pull request by mgarolera**

### CODE REVIEW

File: bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/io/ChannelPoolTest.java [hunk 1]
```diff
@@ -113,8 +112,7 @@ public void testChannelsAreRoundRobinned() throws IOException {
   @Test
   public void testEnsureCapcity() throws IOException {
```
Comment (sduskis) [inline bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/io/ChannelPoolTest.java]: TestCapacity?

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java [hunk 2]
```diff
@@ -54,54 +54,36 @@
...
-      }
+  public ChannelPool(List<HeaderInterceptor> headerInterceptors, ChannelFactory factory, int channelCount)
+      throws IOException {
+    Preconditions.checkArgument(channelCount > 0, "Channel count has to be at least 1.");
+    Preconditions.checkNotNull(headerInterceptors, "Must pass non null headerInterceptors");
+    this.factory = factory;
+    this.headerInterceptors = ImmutableList.copyOf(headerInterceptors);
+
+    ManagedChannel[] channelArray = new ManagedChannel[channelCount];
+    for (int i = 0; i < channelCount; i ++) {
```
Comment (garye) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java]: i ++ -> i++

File: bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java [hunk 3]
```diff
@@ -180,7 +149,7 @@ public boolean isShutdown() {
   @Override
   public boolean isTerminated() {
     boolean allTerminated = true;
-    for (ManagedChannel managedChannel: channels.get()) {
+    for (ManagedChannel managedChannel: channels) {
```
Comment (sduskis) [inline bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java]: nit: 'managedChannel: ' -> 'managedChannel : '.  There should be a space after managedChannel.

**Comment by sduskis** [comment]
lgtm

**Pull request merged**